data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessorConfig.java

```diff
@@ -31,9 +41,38 @@ public List<MappingsParameterConfig> getMappingsParameterConfigs() {
         return mappingsParameterConfigs;
     }
 
-    @AssertTrue(message = "Either mappings or file_path option needs to be configured.")
+    public List<MappingsParameterConfig> getCombinedParameterConfigs() {
+        if(Objects.isNull(combinedParameterConfigs)){
+            combinedParameterConfigs = mappingsParameterConfigs;
+        }
+        return combinedParameterConfigs;
+    }
+
+    @AssertTrue(message = "Please ensure that at least one of the options, either \"mappings\" or \"file_path\", is properly configured.")
     public boolean hasMappings() {
-        return Stream.of(mappingsParameterConfigs, filePath).filter(n -> n != null).count() != 0;
+        return (Objects.nonNull(mappingsParameterConfigs) && !mappingsParameterConfigs.isEmpty()) || Objects.nonNull(filePath);
+    }
+
+    @AssertTrue(message = "\"mappings\" option should not be empty.")
```

Comment:
Is the `@NotEmpty` annotation not working?

Reply:
It works. But in this particular case, the pipeline should succeed even if only `file_path` option is provided. `@NotEmpty` annotation checks if the field is either empty or null. So if the `mappings` option is not provided it fails, even if `file_path` option is provided. This check I provided is required because if the user configures `mappings:` option but there are no sub-options configured, the pipeline should fail. Like in the scenario below:
 ``` 
  processor:
    - translate:
        file_path: "/path/to/file.yaml"
        mappings:
        
  ```